src/core/midiPorts.cpp

```diff
@@ -468,4 +468,28 @@ void midiReceive(const MidiMsg& mm, const std::string& recipient)
 }
 
 
+/* -------------------------------------------------------------------------- */
+
+std::string shortName(const std::string& port) {
+
+	std::string o = port;
+
+	// Remove non-alphanumeric chars
+	o.erase(std::remove_if(o.begin(), o.end(), 
+	[] (auto const& c) -> bool {return !std::isalnum(c);}), o.end());
+
+	// "y" is a vowel in most languages, let's just keep it, aight?
+	std::string vowels = "aeiouyAEIOUY";
+
+	o.erase(std::remove_if(o.begin(), o.end(), 
+	[&] (auto const& c) -> 
+	bool {return (vowels.find(c)!=std::string::npos);}), o.end());
+
+	if (o.length() > 7)
```

Comment:
Where does this magical 7 comes from.
A forgotten `constexpr auto` somewhere?

Reply:
7 seems to work pretty well for a few cases that I tried. The point is to represent MIDI bindings in multi-port setting in a meaningful way, without a 70-char long port names and cluttered MIDI learn window.
This implementation might change in the near future - still, JACK puts "a2j:" prefix to a port name, for example. I just wanted something that works for now and polish that later.